Submission.java

```diff
@@ -464,7 +464,7 @@ public int getSoundBlocksForStage()
      * @param scriptName 
      * @return category, null if no mapping
      */
-    private String getCategory(String scriptName)
+    protected static String getCategory(String scriptName)
     {
```

Comment:
Since the Sprite class needs to be able to access the HashMap to figure out what category each of its scripts/blocks are in to count them, and I didn't want to duplicate the HashMap and getCategory methods in the Sprite class, I made them protected and static, so the Sprite class can use them too. Or do you think we should create a "Categorizable" interface and put these fields/methods in that and have Submission and Sprite implement it? I wasn't sure which was the simplest/better design...